megamek/src/megamek/common/actions/PunchAttackAction.java

```diff
@@ -295,14 +295,10 @@ else if (!Compute.isInArc(ae.getPosition(), ae.getSecondaryFacing(),
         } else if (hasClaws) {
             toHit.addModifier(1, "Using Claws");
         }
-        
-        if (ae.hasQuirk(OptionsConstants.QUIRK_POS_BATTLE_FIST_LA)
-                && hasHandActuator) {
-            toHit.addModifier(-1, "Battlefist");
-        }
-        
-        if (ae.hasQuirk(OptionsConstants.QUIRK_POS_BATTLE_FIST_RA)
-                && hasHandActuator) {
+
+        if (hasHandActuator
+                && (((arm == PunchAttackAction.RIGHT) && ae.hasQuirk(OptionsConstants.QUIRK_POS_BATTLE_FIST_RA))
+                || ((arm == PunchAttackAction.LEFT) && ae.hasQuirk(OptionsConstants.QUIRK_POS_BATTLE_FIST_RA)))) {
```

Comment:
Almost. Left arm should depend on left arm battlefist quirk.

Reply:
Also, let's consider the "BOTH" punch attack action.

Reply:
I think I'm done coding for the day, too many typos in the last little bit 😂